setuptools/command/bdist_wheel.py

```diff
@@ -597,3 +595,18 @@ def adios(p: str) -> None:
             shutil.copy(license_path, os.path.join(distinfo_path, filename))
 
         adios(egginfo_path)
+
+    def _zip_compression(self) -> int:
+        if (
+            isinstance(self.compression, int)
+            and self.compression in self.supported_compressions.values()
+        ):
+            return self.compression
+
+        try:
+            if isinstance(self.compression, str):
+                return self.supported_compressions[self.compression]
+        except KeyError:
+            pass
```

Comment:
This could be done w/o a try-except.

```suggestion
        if isinstance(self.compression, str):
            compression = self.supported_compressions.get(self.compression)
            if compression is not None:
                return compression
```

And just for fun, using the walrus operator (which I don't like)
```suggestion
        if isinstance(self.compression, str) and (
            compression := self.supported_compressions.get(self.compression)
        ):
            return compression
```